staging/src/k8s.io/client-go/tools/clientcmd/api/helpers.go

```diff
@@ -93,13 +95,10 @@ func ShortenConfig(config *Config) {
 	// http://golang.org/pkg/encoding/json/#Marshal using base64 to encode []byte
 	for key, authInfo := range config.AuthInfos {
 		if len(authInfo.ClientKeyData) > 0 {
-			authInfo.ClientKeyData = redactedBytes
+			authInfo.ClientKeyData = dataOmittedBytes
 		}
 		if len(authInfo.ClientCertificateData) > 0 {
-			authInfo.ClientCertificateData = redactedBytes
-		}
-		if len(authInfo.Token) > 0 {
```

Comment:
Why this conditional is removed?

Reply:
This is within the ShortenConfig() function, which with this PR would no longer be in control of the token field and would instead be in charge of simply shortening the config by omitting the base64 encoded certificate data. The Token field is switched to the expected `REDACTED` output later by the RedactSecrets function if it is required by the flags the user has set.

Reply:
Understood, I've reverted this conditional change and updated the tests to reflect this. I will update the PR description to incorporate this change and the new flag and reversion of `--raw` to it's original functionality.